content/tutorials/first-aws-environment.md

```diff
@@ -0,0 +1,210 @@
+
+```bash
+aws-vault exec $YOUR_PROFILE -- atmos terraform plan tfstate-backend --stack ue2-root
+
+# Check the plan looks good and doesn't have 'TODO' anywhere if so, be sure to run `bin/uniquify.sh`)
+aws-vault exec $YOUR_PROFILE -- atmos terraform apply tfstate-backend --stack ue2-root
+```
+
+This will provision our S3 bucket + Dynamo DB table for usage as our backend.
+
+3. Next, let's generate the configuration for that backend:
+
+```bash
+atmos terraform backend generate component tfstate-backend --stack ue2-root
+```
+
+This will look at our stack, find the imported `terraform.backend.s3` configuration and build a valid `backend.tf.json` file and put it in our component directory. Then going forward, whenver we `plan` or `apply` against that component it will be properly configured to use that S3 backend.
+
+4. Now that we have our `backend.tf.json` file, we can change over our `tfstate-backend` component from using local state to its S3 backend:
+
+```bash
+aws-vault exec $YOUR_PROFILE -- atmos terraform plan tfstate-backend --stack ue2-root
+```
+
+This will prompt you with the following:
+
+```txt
+Do you want to migrate all workspaces to "s3"?
+Both the existing "local" backend and the newly configured "s3" backend
+support workspaces. When migrating between backends, Terraform will copy
+all workspaces (with the same names). THIS WILL OVERWRITE any conflicting
+states in the destination.
+
+Terraform initialization doesn't currently migrate only select workspaces.
+If you want to migrate a select number of workspaces, you must manually
+pull and push those states.
+
+If you answer "yes", Terraform will migrate all states. If you answer
+"no", Terraform will abort.
+
+Enter a value:
+```
+
+Enter "yes" and this will migrate our local state to our S3 backend. Success!
+
+5. Now that we've migrated our single "chicken or the egg" component, we'll want to generate the `backend.tf.json` file for our other components as well considering we'll want them to use the S3 backend too. Luckily for us, there is only one, simple component as part of this tutorial: `components/terraform/static-site`. Let's go ahead and generate the backend config for that component:
+
+```bash
+atmos terraform backend generate component static-site --stack uw2-dev
+```
+
+Exactly the same as for our `tfstate-backend` component, our `atmos` `backend generate` command will put a `backend.tf.json` file in our `components/terraform/static-site/` directory so that component will always utilize the correct backend. If this were a real project that we were working on then we'd actually check those files into git so that our backend configuration would persist going forward, but since this is a tutorial we'll skip that step.
```

Comment:
```suggestion
Exactly the same as for our `tfstate-backend` component, our `atmos` `backend generate` command will put a `backend.tf.json` file in our `components/terraform/static-site/` directory so that component will always utilize the correct backend. If this were a real project that we were working on and practicing GitOps, then we'd actually check those files into git so that our backend configuration would persist going forward, but since this is a tutorial we'll skip that step.
```